include/swift/AST/DiagnosticsSema.def

```diff
@@ -663,6 +663,9 @@ ERROR(super_initializer_not_in_initializer,none,
       "'super.init' cannot be called outside of an initializer", ())
 WARNING(isa_is_always_true,none, "'%0' test is always true",
         (StringRef))
+WARNING(isa_is_foreign_check,none,
+      "'%0' test always succeeds because %1 is a foreign type",
```

Comment:
Ah, maybe we don't diagnose for `as!` because there's no way to silence that diagnostic and it's sometimes useful…whereas this one isn't.